File: js solutions/problem2.js
var fibSum = 0
var fibArray = [0, 1]
var i = 1
while (i < 4000000) {
    fibArray.push(i)
    i = fibArray[(fibArray.length - 1)] + fibArray[(fibArray.length - 2)]
    if (i % 2 == 0){
        fibSum = fibSum + i
        console.log(fibSum)
    }
}

File: kt solutions/problem2.kt
import java.util.Arrays
fun main(){
    var fibSum: Int = 0
    var curFib: Int = 0
    var prevFib1: Int = 1
    var prevFib2: Int = 0
    while(curFib < 4000000){

        curFib = prevFib1 + prevFib2
        println(curFib)

        prevFib2 = prevFib1
        prevFib1 = curFib

        if (curFib.rem(2)==0){
            fibSum = fibSum + curFib
        }
    }
    println("sum of even fibonacci numbers under 4,000,000: $fibSum")
}
